packages/react-components/src/components/video-player/videoPlayer.tsx

```diff
@@ -61,6 +61,8 @@ export const VideoPlayer = (props: IVideoPlayerProps) => {
   const [playbackMode, setPlaybackMode] = useState<typeof PLAYBACKMODE_ON_DEMAND | typeof PLAYBACKMODE_LIVE>(
     initialPlaybackMode
   );
+  const playbackModeRef = useRef({});
+  playbackModeRef.current = playbackMode;
```

Comment:
why do we need this? it seems to be overly complex to have playback in props, states and also ref

Reply:
it doesn't seem like there are code needs to be executed when the playbackMode state is changed except `updateVideoSource`, you can probably just delete the state and call updateVideoSource when toggle the mode

Reply:
videoPlayerReady() is a callback which is called when video player is initialized. Live button is added to the control bar here with onClick(). Problem was that the event was running in its own closure and had the value at the time when the event was bounded. To fix this, I had to use useRef().

Reply:
I made the changes to remove the state var for playbackMode and calling the updateVideoSource() directly when toggling the playbackMode and it is not working. The instance of VideoPlayer is null in that case. I think it is because of the context in which the call is made. The instance is video player in case of this call is not present.

Currently, it is working because on toggle I change the state and the state change for a component variable updates the component i.e. the current video player.